src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
+
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_VALUE = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int RESIZE_INDEX = 2;
+    private Node [] elements = new Node[DEFAULT_VALUE];
+    private int size;
+
+    private class Node<K,V> {
+        private Integer hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > elements.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHashCode(key) % elements.length;
+        if (elements[index] != null) {
+            Node currentBucket = elements[index];
+            do {
+                if (currentBucket.hash == getHashCode(key)
+                        && Objects.equals(key, currentBucket.key)) {
+                    currentBucket.value = value;
+                    return;
+                }
+                if (currentBucket.next != null) {
+                    currentBucket = currentBucket.next;
+                }
+            } while (currentBucket.next != null);
+            if (currentBucket.hash == getHashCode(key) && Objects.equals(key, currentBucket.key)) {
+                currentBucket.value = value;
+                return;
+            }
+            Node newNode = new Node<>(getHashCode(key), key, value, null);
+            currentBucket.next = newNode;
+            size++;
+            return;
+        }
+        elements[index] = new Node(getHashCode(key), key, value, null);
+        size++;
```

Comment:
too complex logic, let's simplify it:
1. Find bucket by index
2. if bucket == null fill this bucket with a new node
3. else use a loop and go through all chain of this bucket, if key equals change the value and use return;
4. if you get to the moment when the next element of the current node is null fill this next element with a new node

Reply:
OK, let's talk about it on a Q&A